src/node/plugin.ts

```diff
@@ -195,7 +195,7 @@ export async function createVitePressPlugin(
     },
 
     renderStart() {
-      if (hasDeadLinks && !ignoreDeadLinks) {
+      if (hasDeadLinks && ignoreDeadLinks !== true) {
```

Comment:
This can be removed IG 👀  (since you're now handling this in `shouldIgnoreDeadLink` itself). `hasDeadLinks` will be false.

```suggestion
      if (hasDeadLinks) {
```